### ISSUE TITLE
Add startPoint option to control where initial words get placed

### ISSUE THREAD

**Issue by cesine** [iLanguage/d3-cloud#31]
<a href="https://github.com/davidjb"><img src="https://avatars.githubusercontent.com/u/1002811?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [davidjb](https://github.com/davidjb)**
_Monday Jan 14, 2013 at 06:17 GMT_
_Originally opened as https://github.com/jasondavies/d3-cloud/pull/8_

----

This adds the ability to control the starting position ([x,y] array) for the initial word/term being placed in the word cloud and can be used like so:

```
var width = 600;
var height = 300;
d3.layout.cloud().size([width, height])
          .startPoint([width/2.0, height/2.0])
...
```
which would ensure the first term is placed at the centre of the cloud. The startPoint can be any x,y position, however.

The parameter name could be something better or more descriptive, but the idea is there.

----

_**[davidjb](https://github.com/davidjb)** included the following code: https://github.com/jasondavies/d3-cloud/pull/8/commits_

**Comment by cesine** [comment]
<a href="https://github.com/holgerd77"><img src="https://avatars.githubusercontent.com/u/931137?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [holgerd77](https://github.com/holgerd77)**
_Friday Mar 22, 2013 at 09:09 GMT_

----

Hi, I would support this patch. I have got a use case, where I use the tag cloud as a concrete navigation element for document search and there it is confusing when the tag cloud looks different every time the page is reloaded. Therefore I would also support an additional switch `staticLayout`, which eliminates random layouting if desired.

I have done this manually at the moment, by using the patch above, setting start point and replacing line 97/98 with:

dt = 0.43096132576465607 < .5 ? 1 : -1, // replaced to get static, predictable tag cloud
//dt = Math.random() < .5 ? 1 : -1,

I think, semantics together with the startPoint attribute would make sense the following way:
if no startPoint is set, a random static number is used, hard coded like in the two lines above. If a startPoint is set, it is used and only the random function above is exchanged with a static number.

Otherwise: thanks for the great work, can't wait to see my tag cloud in production! :-)

**Comment by cesine** [comment]
<a href="https://github.com/davidjb"><img src="https://avatars.githubusercontent.com/u/1002811?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [davidjb](https://github.com/davidjb)**
_Monday Mar 25, 2013 at 22:34 GMT_

----

@holgerd77 Yes, that's something I'd be very much interested in as well.  I'm not too fussed in the way to achieve it, but a replicable tag cloud would be excellent. Hence I was interested in #1.

**Comment by cesine** [comment]
duplicate import with https://github.com/IQAndreas/github-issues-import

merged in #70
